### PR TITLE
Replace FastList.newList(iterable) with Lists.mutable.withAll().

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/CompositeFastList.java [hunk 1]
```diff
@@ -238,7 +239,7 @@ public boolean addAll(Collection<? extends E> collection)
             return false;
         }
         Collection<? extends E> collectionToAdd =
-                collection instanceof FastList ? collection : FastList.newList(collection);
+                collection instanceof FastList ? collection : Lists.mutable.withAll(collection);
```
Comment (donraab) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/CompositeFastList.java:242]: I think this needs to be left as `FastList` as the implementation must be a `FastList`.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MultiReaderFastList.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public static <T> MultiReaderFastList<T> newList(int capacity)
 
     public static <T> MultiReaderFastList<T> newList(Iterable<? extends T> iterable)
     {
-        return new MultiReaderFastList<>(FastList.newList(iterable));
```
Comment (donraab) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MultiReaderFastList.java:129]: I think this should be left as a `FastList` as this is a `MultiReaderFastList`.

**Review by donraab — CHANGES_REQUESTED** [review]
A few suggestions inline.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/CompositeFastList.java (hunk 1 above)
Comment (motlin) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/CompositeFastList.java]: Discarded this change.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MultiReaderFastList.java (hunk 2 above)
Comment (motlin) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MultiReaderFastList.java]: Discarded this change to MultiReaderFastList.

**Review by donraab — APPROVED** [review]
LGTM. I'll merge once builds complete. Thanks for the contribution @motlin !

**Pull request merged**